### ISSUE TITLE
note the convention of using 'self' for closure variable name

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/phet-info#30]
@jbphet will note the convention of using 'self' for closure variable name in the developer guide and code review checklist.  

From 9/15/16 developer meeting notes:

> ✓JB: What is current thinking on using var self = this vs. var descriptiveName = this?  I believe it’s developer choice based on what I see in the code, and I’d like to make sure we are all still okay with this.
SR: using var “self” shadows an existing javascript “self” variable.  For a class of MyComponent, I typically use var myComponent = this;  This works well for me and I am habituated to it, but not opposed to switching to use var thisMyComponent = this; if that is clearer for everyone else.
AA: the global SR mentions is window.self: https://developer.mozilla.org/en-US/docs/Web/API/Window/self
CM: I prefer the “this” prefix because it identifies it as a context variable. I’ve been using thisNode, thisModel, thisScreenView, etc. And I’ve been resisting the urge to change “self” in sims that I’ve worked on.
CM: 489 occurrences of “self =”, 1609 occurrences of “= this;”
JG: I prefer “self” because it is a consistent variable name for any and all objects that need extended scope.  It saves me from looking through the constructor for a unique variable name every time I need to carry this through a different context.
DS: Every other place I’ve worked with using Javascript uses “self”.
AA: I have been using `self`, because it’s consistent, the context is usually very clear without custom-naming it, and it’s short. I think it’s not so nice that there’s no convention for it. Most other things like this have been standardized, and this stands out. 
SR: I saw some are using _this
CM: This is vestigial from privateness.  Only use underscore if you need to implement a ES5 set/get.
**Consensus: Use self going forward, retrofit code as needed. JB will note in developer guide and code review checklist.**

**Comment by pixelzoom** [comment]
Developer Guide: http://bit.ly/phet-development-overview

Code review checklist: https://github.com/phetsims/phet-info/blob/master/checklists/code_review_checklist.md

**Comment by jbphet** [comment]
I've added a bullet item for this to the PhET Development Overview.  I looked at the Code Review Checklist, and it didn't seem to me like a separate checklist item was warranted.  I discussed this with @pixelzoom and @samreid via Skype, and they agreed, especially since there is a new lint rule that will check for appropriate use of ```self``` versus ```this```.  So, this should now be addressed.  Closing.
